components/public-api-server/pkg/apiv1/identityprovider.go

```diff
@@ -71,12 +72,15 @@ func (srv *IdentityProviderService) GetIDToken(ctx context.Context, req *connect
 	}
 
 	var email string
+	var lastSigninTime time.Time
 	for _, id := range user.Identities {
 		if id == nil || id.Deleted || id.PrimaryEmail == "" {
 			continue
 		}
-		email = id.PrimaryEmail
-		break
+		if email == "" || id.LastSigninTime.After(lastSigninTime) {
+			email = id.PrimaryEmail
+			lastSigninTime = id.LastSigninTime
+		}
 	}
```

Comment:
For readability, I recommend pulling this into a separate function where you return the moment you find the first valid set. The use of the `email == "" || ...` as a continuation is confusing, and non-obvious when scanning the code